### PR TITLE
Feat: support more Sorald options

### PR DESCRIPTION
**Pull request by henry-lp**
Signed-off-by: henry-lp <tailp@kth.se>

### CODE REVIEW

**Comment by henry-lp** [comment]
Waiting for latest version of Sorald to be deployed.

**Comment by fermadeiral** [comment]
> Waiting for latest version of Sorald to be deployed.

https://oss.sonatype.org/content/repositories/snapshots/se/kth/castor/sorald/

**Comment by monperrus** [comment]
Last version from Mon Sep 28 11:49:02

**Comment by henry-lp** [comment]
ready for merge :) . After this I'll also make a PR to update Jenkins Plugin.

File: src/pom.xml [hunk 1]
```diff
@@ -55,9 +55,8 @@
         <module>repairnator-pipeline</module>
         <module>repairnator-core</module>
         <module>repairnator-realtime</module>
-				<module>repairnator-jenkins-plugin</module>
```
Comment (monperrus) [inline src/pom.xml:58]: why do you remove the jenkins plugin?

File: src/repairnator-core/src/main/java/fr/inria/spirals/repairnator/config/RepairnatorConfig.java [hunk 2]
```diff
@@ -54,6 +54,11 @@ public String getKlass() {
         OVERFITTING
     }
 
+    public enum SORALD_REPAIR_MODE {
+        DEFAULT,
```
Comment (monperrus) [inline src/repairnator-core/src/main/java/fr/inria/spirals/repairnator/config/RepairnatorConfig.java:58]: add comment to explain the meaning of each enum option

File: src/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/repair/Sorald.java [hunk 3]
```diff
@@ -62,17 +62,18 @@ protected StepStatus businessExecute() {
         for (String rule : RepairnatorConfig.getInstance().getSonarRules()) {
             this.getLogger().info("Repo: " + pathToRepoDir);
             this.getLogger().info("Try to repair rule " + rule);
-
-            try {
-                Main.main(new String[]{
+            String[] args = new String[]{
                             "--originalFilesPath",pathToRepoDir,
                             "--ruleKeys",rule,
-                            "--workspace",RepairnatorConfig.getInstance().getWorkspacePath(),
```
Comment (monperrus) [inline src/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/repair/Sorald.java:70]: excellent, always good to replace global state by local state

File: src/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/repair/Sorald.java [hunk 4]
```diff
@@ -62,17 +62,18 @@ protected StepStatus businessExecute() {
...
                             "--gitRepoPath",pathToRepoDir,
                             "--prettyPrintingStrategy","SNIPER",
-                            "--maxFixesPerRule","1"});
+                            "--maxFixesPerRule","" + getConfig().getSoraldMaxFixesPerRule(),
+                            "--repairStrategy",RepairnatorConfig.getInstance().getSoraldRepairMode().name(),
+                            "--maxFilesPerSegment","" + RepairnatorConfig.getInstance().getSegmentSize()};
+            try {
+                Main.main(args);
             } catch(Exception e) {
-                e.printStackTrace();
```
Comment (monperrus) [inline src/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/repair/Sorald.java:75]: use proper call to the logging library

**Review by monperrus — COMMENTED** [review]
Thanks a lot got the update Henry, see comments

File: src/pom.xml (hunk 1 above)
Comment (henry-lp) [reply src/pom.xml:58]: I got warning about the pom being malformed due to the Jenkins plugin pom has another parent pom than repairnator-core like other projects. I suspect it also caused a bug discovered when I tried to update Sorald some time ago.

Comment (monperrus) [reply src/pom.xml:58]: The problem is that if we remove this, CI does not compile and test the jenkins plugin anymore.

The warning is not important, we can put this line back.

Comment (henry-lp) [reply src/pom.xml:58]: Then I'll put it back for now. If there are any issue, we can just have a travis build step for the Jenkins with custom script different from the other build step. :)

File: src/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/repair/Sorald.java (hunk 4 above)
Comment (henry-lp) [reply src/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/repair/Sorald.java:75]: `e.printStackTrace()` is already replaced by `StepStatus.buildSkipped(..)` in the subsequent line.

**Comment by monperrus** [comment]
Thanks a lot @henry-lp 

Could you make a PR to document the Sorald support in the docs folder? Thanks!

**Comment by henry-lp** [comment]
@monperrus yes :) ,

**Pull request merged**